### ISSUE TITLE
stickybit and multiple groups behaviour : wrong File ownership with v1.6.1 and file creation impossible with v1.7-2

### ISSUE THREAD

**Issue by Mamak2000** [rfjakob/gocryptfs#434]
Hello,

I tested gocryptfs versions 1.2-2 (default package on raspbian), 1.6.1 (gocryptfs_1.6.1-1+b20_armhf.deb) taken from debian packages for ARMHF and 1.7-2 (gocryptfs_1.7-2_armhf.deb).
I tried both gocryptfs command lines to mount the encrypted partition:
```
 sudo gocryptfs -aessiv -allow_other -ko suid /NAS/.gocryptfs_chiffre/ /NAS/VISIBLE/donnes_gocryptfs
 sudo gocryptfs -aessiv -allow_other /NAS/.gocryptfs_chiffre/ /NAS/VISIBLE/donnes_gocryptfs
```
This gives the following result with the mount command :
```
 pi@raspnas:~ $ mount
/NAS/.gocryptfs_chiffre on /NAS/VISIBLE/donnes_gocryptfs type fuse.gocryptfs (rw,nodev,relatime,user_id=0,group_id=0,default_permissions,allow_other,max_read=131072)
```
### Observed behaviour:
The results are always the same. To sum up,
 - until version 1.6.1, file creation/edition works but the group ownership is wrong.
 - As of version 1.7-2, the group ownership is correct BUT files cannot be created in subdirectories that belong to secondary groups. I encounter the same issue with encfs 1.9.1 till 1.9.5

On a plaintext directory structure (without encryption), this works perfectly fine.

### Questions:
Am I missing something? (wrong mount options ?)
Are there any workaround ?
There seems to be a kind of regression that comes with the proper group ownership setting, doesn't it ? Could it be related to the Setreuid that was introduced in version 1.7 ?
Do you need any further debugs/explanations ?

### Plot:
I have a directory structure with several users who belong to various groups :

- user "louis" belongs to famille + parents + enfants + eleonore + apolline
- user "ag" belongs to famille + parents + enfants + eleonore + apolline
- user "eleonore" belongs to famille + enfants + eleonore
- user "apolline" belongs to famille + enfants + apolline
- group "famille" is the primary group for the users. Others are secondary-groups.

Example for user louis:	
```
$ id -a
uid=1002(louis) gid=1001(famille) groupes=1001(famille),46(plugdev),1002(parents),1003(enfants),1006(eleonore),1007(apolline)
```

I am using raspbian
```
pi@raspnas:/var/tmp $ uname -a
Linux raspnas 4.19.66-v7+ #1253 SMP Thu Aug 15 11:49:46 BST 2019 armv7l GNU/Linux
```

I set up several directories as follows (note the stickybits):
```
pi@raspnas:~ $ ls -al /NAS/VISIBLE/
total 36
drwxr-xr-x  9 root root    4096 nov.   3 10:34 .
drwxr-xrwx 12 root root    4096 nov.   3 10:31 ..
drwxrws---  7 pi   famille 4096 nov.   3 22:20 donnes_gocryptfs

pi@raspnas:~ $ ls -al /NAS/VISIBLE/donnes_gocryptfs
total 40
drwxrws--- 7 pi       famille  4096 nov.   3 22:20 .
drwxr-xr-x 9 root     root     4096 nov.   3 10:34 ..
drwxrws--- 3 ag       parents  4096 nov.   3 22:29 ag
drwxrws--- 2 eleonore eleonore 4096 nov.   3 10:44 eleonore
drwxrws--- 2 eleonore enfants  4096 oct.  30 18:27 enfants
-rw-rw---- 1 ag       famille    17 oct.  22 21:34 fichier_ag.txt
-rw-rw---- 1 eleonore famille     5 oct.  22 21:31 fichier_eleonore.txt
-rw-rw---- 1 louis    famille     3 oct.  22 21:30 lkkllk.txt
drwxrws--- 2 louis    parents  4096 nov.   3 22:28 louis
drwxrws--- 2 louis    parents  4096 oct.  22 21:34 parents
```

### Until version 1.6.1 included
Until version 1.6.1 included user can edit/create files in the directories provided that they belong to the correct group BUT the files/directories ownership are not properly set. They are always created with the user's primary group instead of abiding by the stickybit that would force the proper group.
Namely, if user louis wants to create a file in "ag" directory, then, the file is created with ownership "louis:famille" instead of "louis:parents"

```
$ touch /NAS/VISIBLE/donnes_gocryptfs/louis/fichier_louis_1.6.1.txt
$ touch /NAS/VISIBLE/donnes_gocryptfs/ag/fichier_louis_1.6.1.txt
```
```
$ ls -al  /NAS/VISIBLE/donnes_gocryptfs/louis
total 8720
drwxrws--- 2 louis    parents    4096 nov.   3 22:28 .
drwxrws--- 7 pi       famille    4096 nov.   3 22:20 ..
-rw-rw---- 1 louis    parents      20 oct.  22 21:18 essai.txt.txt
-rw-rw---- 1 eleonore parents      16 oct.  22 21:33 fichier_eleonore.txt
-rw-rw---- 1 louis    famille       0 nov.   3 22:28 fichier_louis_1.6.1.txt  <== wrong group
-rw-rw---- 1 louis    parents       0 oct.  31 16:58 fichier_louis_test2.txt
-rw-rw---- 1 louis    parents       0 oct.  31 10:09 fichier_louis_test.txt
```
wrong group for the new file

```
$  ls -al  /NAS/VISIBLE/donnes_gocryptfs/ag
total 20
drwxrws--- 3 ag    parents 4096 nov.   3 22:29 .
drwxrws--- 7 pi    famille 4096 nov.   3 22:20 ..
-rw-rw-r-- 1 ag    parents    0 oct.  30 18:35 fichier_ag_20191030.txt
-rw-rw---- 1 ag    parents   50 nov.   3 21:01 fichier_ag.txt
-rw-rw---- 1 louis famille    0 nov.   3 22:29 fichier_louis_1.6.1.txt  <== wrong group
-rw-rw-r-- 1 ag    parents   27 oct.  31 09:34 fichier_maman.txt
drwxrws--- 2 ag    parents 4096 nov.   3 21:09 sous-rep_ag
```
Still wrong group for the new file

**Debug is the following:**
```
pi@raspnas:/var/tmp $ sudo gocryptfs -aessiv -allow_other -d -fg -fusedebug -ko suid /NAS/.gocryptfs_chiffre/ /NAS/VISIBLE/donnes_gocryptfs
OpenSSL enabled
cli args: &main.argContainer{debug:true, init:false, zerokey:false, fusedebug:true, openssl:true, passwd:false, fg:true, version:false, plaintextnames:false, quiet:false, nosyslog:false, wpanic:false, longnames:true, allow_other:true, reverse:false, aessiv:true, nonempty:false, raw64:true, noprealloc:false, speed:false, hkdf:true, serialize_reads:false, forcedecode:false, hh:false, info:false, sharedstorage:false, devrandom:false, fsck:false, trezor:false, dev:false, nodev:false, suid:false, nosuid:false, exec:false, noexec:false, rw:false, ro:false, masterkey:"", mountpoint:"/NAS/VISIBLE/donnes_gocryptfs", cipherdir:"/NAS/.gocryptfs_chiffre", cpuprofile:"", extpass:"", memprofile:"", ko:"suid", passfile:"", ctlsock:"", fsname:"", force_owner:"", trace:"", exclude:main.multipleStrings(nil), config:"/NAS/.gocryptfs_chiffre/gocryptfs.conf", notifypid:0, scryptn:16, _configCustom:false, _ctlsockFd:net.Listener(nil), _forceOwner:(*fuse.Owner)(nil)}
Password:
Decrypting master key
frontendArgs: {
        "Cipherdir": "/NAS/.gocryptfs_chiffre",
        "PlaintextNames": false,
        "LongNames": true,
        "PreserveOwner": true,
        "ForceOwner": null,
        "ConfigCustom": false,
        "NoPrealloc": false,
        "SerializeReads": false,
        "ForceDecode": false,
        "Exclude": null
}
The option "-allow_other" is set. Make sure the file permissions protect your data from unwanted access.
Adding -ko mount options: [suid]
Filesystem mounted and ready.

2019/11/03 22:28:01 rx 2: GETATTR i1 {Fh 0}
FS.GetAttr('')
encryptPath '' -> '' (err: <nil>)
2019/11/03 22:28:01 tx 2:     OK, {tA=1s {M042770 SZ=4096 L=7 1000:1001 B8*4096 i0:1151489 A 1572816037.663566 M 1572816036.633572 C 1572816036.633572}}
2019/11/03 22:28:01 rx 3: LOOKUP i1 ["louis"] 6b
FS.GetAttr('louis')
encryptPath 'louis' -> 'W7e9_jVG120Qn2GnAwPRfw==' (err: <nil>)
2019/11/03 22:28:01 tx 3:     OK, {i3 g2 tE=1s tA=1s {M042770 SZ=4096 L=2 1002:1002 B8*4096 i0:1151502 A 1572816112.113162 M 1572816329.441979 C 1572816329.441979}}
2019/11/03 22:28:01 rx 4: LOOKUP i3 ["fichier_louis_1.6.1.txt"] 24b
FS.GetAttr('louis/fichier_louis_1.6.1.txt')
encryptPath 'louis/fichier_louis_1.6.1.txt' -> 'W7e9_jVG120Qn2GnAwPRfw==/gP8dDTZnda6Ok75-ycxPwW6aYZvMFw1Ye5sPgobTHAo=' (err: <nil>)
FS.GetAttr failed: 2=no such file or directory
2019/11/03 22:28:01 tx 4:     OK, {i0 g0 tE=1s tA=0s {M00 SZ=0 L=0 0:0 B0*0 i0:0 A 0.000000 M 0.000000 C 0.000000}}
2019/11/03 22:28:01 rx 5: CREATE i3 {0100660 [LARGEFILE,NONBLOCK,WRONLY,CREAT] (07)} ["fichier_louis_1.6.1.txt"] 24b
encryptPath 'louis/fichier_louis_1.6.1.txt' -> 'W7e9_jVG120Qn2GnAwPRfw==/gP8dDTZnda6Ok75-ycxPwW6aYZvMFw1Ye5sPgobTHAo=' (err: <nil>)
getBackingPath: /NAS/.gocryptfs_chiffre + louis/fichier_louis_1.6.1.txt -> /NAS/.gocryptfs_chiffre/W7e9_jVG120Qn2GnAwPRfw==/gP8dDTZnda6Ok75-ycxPwW6aYZvMFw1Ye5sPgobTHAo=
FS.GetAttr('louis/fichier_louis_1.6.1.txt')
encryptPath 'louis/fichier_louis_1.6.1.txt' -> 'W7e9_jVG120Qn2GnAwPRfw==/gP8dDTZnda6Ok75-ycxPwW6aYZvMFw1Ye5sPgobTHAo=' (err: <nil>)
2019/11/03 22:28:01 tx 5:     OK, {i4 g3 {M0100660 SZ=0 L=1 1002:1001 B0*4096 i0:1151534 A 1572816481.101154 M 1572816481.101154 C 1572816481.101154} &{2 0 0}}
2019/11/03 22:28:01 rx 6: FLUSH i4 {Fh 2}
2019/11/03 22:28:01 tx 6:     OK
2019/11/03 22:28:01 rx 7: SETATTR i4 {atime 1572816481.101154380, mtime 1572816481.101154380}
file.GetAttr()
2019/11/03 22:28:01 tx 7:     OK, {tA=1s {M0100660 SZ=0 L=1 1002:1001 B0*4096 i0:1151534 A 1572816481.105289 M 1572816481.105293 C 1572816481.101154}}
2019/11/03 22:28:01 rx 8: FLUSH i4 {Fh 2}
2019/11/03 22:28:01 tx 8:     OK
2019/11/03 22:28:01 rx 9: RELEASE i4 {Fh 2 NONBLOCK,LARGEFILE,WRONLY  L0}
2019/11/03 22:28:01 tx 9:     OK

2019/11/03 22:29:02 rx 10: GETATTR i1 {Fh 0}
FS.GetAttr('')
encryptPath '' -> '' (err: <nil>)
2019/11/03 22:29:02 tx 10:     OK, {tA=1s {M042770 SZ=4096 L=7 1000:1001 B8*4096 i0:1151489 A 1572816037.663566 M 1572816036.633572 C 1572816036.633572}}
2019/11/03 22:29:02 rx 11: LOOKUP i1 ["ag"] 3b
FS.GetAttr('ag')
encryptPath 'ag' -> '9kAVe84V1Zp9BsmJlZkm_g==' (err: <nil>)
2019/11/03 22:29:02 tx 11:     OK, {i5 g4 tE=1s tA=1s {M042770 SZ=4096 L=3 1001:1002 B8*4096 i0:1151492 A 1572815646.215696 M 1572815646.135696 C 1572815646.135696}}
2019/11/03 22:29:02 rx 12: LOOKUP i5 ["fichier_louis_1.6.1.txt"] 24b
FS.GetAttr('ag/fichier_louis_1.6.1.txt')
encryptPath 'ag/fichier_louis_1.6.1.txt' -> '9kAVe84V1Zp9BsmJlZkm_g==/Z9nujU-9Gu3RnW_rz_Yz-ShySO75zc7-02LMAGftt7E=' (err: <nil>)
FS.GetAttr failed: 2=no such file or directory
2019/11/03 22:29:02 tx 12:     OK, {i0 g0 tE=1s tA=0s {M00 SZ=0 L=0 0:0 B0*0 i0:0 A 0.000000 M 0.000000 C 0.000000}}
2019/11/03 22:29:02 rx 13: CREATE i5 {0100660 [CREAT,NONBLOCK,LARGEFILE,WRONLY] (07)} ["fichier_louis_1.6.1.txt"] 24b
encryptPath 'ag/fichier_louis_1.6.1.txt' -> '9kAVe84V1Zp9BsmJlZkm_g==/Z9nujU-9Gu3RnW_rz_Yz-ShySO75zc7-02LMAGftt7E=' (err: <nil>)
getBackingPath: /NAS/.gocryptfs_chiffre + ag/fichier_louis_1.6.1.txt -> /NAS/.gocryptfs_chiffre/9kAVe84V1Zp9BsmJlZkm_g==/Z9nujU-9Gu3RnW_rz_Yz-ShySO75zc7-02LMAGftt7E=
FS.GetAttr('ag/fichier_louis_1.6.1.txt')
encryptPath 'ag/fichier_louis_1.6.1.txt' -> '9kAVe84V1Zp9BsmJlZkm_g==/Z9nujU-9Gu3RnW_rz_Yz-ShySO75zc7-02LMAGftt7E=' (err: <nil>)
2019/11/03 22:29:02 tx 13:     OK, {i6 g5 {M0100660 SZ=0 L=1 1002:1001 B0*4096 i0:1151538 A 1572816542.010829 M 1572816542.010829 C 1572816542.010829} &{2 0 0}}
2019/11/03 22:29:02 rx 14: FLUSH i6 {Fh 2}
2019/11/03 22:29:02 tx 14:     OK
2019/11/03 22:29:02 rx 15: SETATTR i6 {atime 1572816542.010829184, mtime 1572816542.010829184}
file.GetAttr()
2019/11/03 22:29:02 tx 15:     OK, {tA=1s {M0100660 SZ=0 L=1 1002:1001 B0*4096 i0:1151538 A 1572816542.017957 M 1572816542.017960 C 1572816542.010829}}
2019/11/03 22:29:02 rx 16: FLUSH i6 {Fh 2}
2019/11/03 22:29:02 tx 16:     OK
2019/11/03 22:29:02 rx 17: RELEASE i6 {Fh 2 NONBLOCK,LARGEFILE,WRONLY  L0}
2019/11/03 22:29:02 tx 17:     OK
```

### As of version 1.7-2
As of version 1.7-2, I am facing the same issue I had on encfs in public mode.
- In directories that belong to "famille" any user can read/write and create files/sub-directories.
- In directories that belong to "ag:parents", only user "ag" can create new files. User "louis" can read the files or modify existing ones but cannot create new files. I assume it is because the directory belongs to user "ag" as well as group "parents" (so since user "ag" owns the directory and therfore there is no permission issue).
- In directories that belong to "eleonore:enfants", only user "eleonore" can create a file. Users "louis" and "ag" can modify existing files and read them but cannot create new ones.

With the same tests of file creation here are the results
```
$ touch /NAS/VISIBLE/donnes_gocryptfs/ag/fichier_louis_1.7-2.txt
touch: impossible de faire un touch '/NAS/VISIBLE/donnes_gocryptfs/ag/fichier_louis_1.7-2.txt': Permission non accordée
$ touch /NAS/VISIBLE/donnes_gocryptfs/louis/fichier_louis_1.7-2.txt
$ ls -al  /NAS/VISIBLE/donnes_gocryptfs/louis/
total 8720
drwxrws--- 2 louis    parents    4096 nov.   3 22:21 .
drwxrws--- 7 pi       famille    4096 nov.   3 22:20 ..
-rw-rw---- 1 louis    parents      20 oct.  22 21:18 essai.txt.txt
-rw-rw---- 1 eleonore parents      16 oct.  22 21:33 fichier_eleonore.txt
-rw-rw---- 1 louis    parents       0 nov.   3 22:21 fichier_louis_1.7-2.txt <== group is right
```

**DEBUG**
```
pi@raspnas:/var/tmp $ sudo gocryptfs -aessiv -allow_other -d -fg -fusedebug -ko suid /NAS/.gocryptfs_chiffre/ /NAS/VISIBLE/donnes_gocryptfs
OpenSSL enabled
cli args: &main.argContainer{debug:true, init:false, zerokey:false, fusedebug:true, openssl:true, passwd:false, fg:true, version:false, plaintextnames:false, quiet:false, nosyslog:false, wpanic:false, longnames:true, allow_other:true, reverse:false, aessiv:true, nonempty:false, raw64:true, noprealloc:false, speed:false, hkdf:true, serialize_reads:false, forcedecode:false, hh:false, info:false, sharedstorage:false, devrandom:false, fsck:false, trezor:false, dev:false, nodev:false, suid:false, nosuid:false, exec:false, noexec:false, rw:false, ro:false, masterkey:"", mountpoint:"/NAS/VISIBLE/donnes_gocryptfs", cipherdir:"/NAS/.gocryptfs_chiffre", cpuprofile:"", memprofile:"", ko:"suid", passfile:"", ctlsock:"", fsname:"", force_owner:"", trace:"", extpass:main.multipleStrings(nil), exclude:main.multipleStrings(nil), config:"/NAS/.gocryptfs_chiffre/gocryptfs.conf", notifypid:0, scryptn:16, idle:0, _configCustom:false, _ctlsockFd:net.Listener(nil), _forceOwner:(*fuse.Owner)(nil)}
Password:
Decrypting master key
CryptoCore.Wipe: Only nil'ing stdlib refs
frontendArgs: {
        "Cipherdir": "/NAS/.gocryptfs_chiffre",
        "PlaintextNames": false,
        "LongNames": true,
        "PreserveOwner": true,
        "ForceOwner": null,
        "ConfigCustom": false,
        "NoPrealloc": false,
        "SerializeReads": false,
        "ForceDecode": false,
        "Exclude": null
}
The option "-allow_other" is set. Make sure the file permissions protect your data from unwanted access.
Adding -ko mount options: [suid]
Filesystem mounted and ready.
2019/11/03 22:24:12 rx 2: GETATTR i1 {Fh 0}
FS.GetAttr("")
2019/11/03 22:24:12 tx 2:     OK, {tA=1s {M042770 SZ=4096 L=7 1000:1001 B8*4096 i0:1151489 A 1572816037.663566 M 1572816036.633572 C 1572816036.633572}}
2019/11/03 22:24:12 rx 3: LOOKUP i1 ["ag"] 3b
FS.GetAttr("ag")
2019/11/03 22:24:12 tx 3:     OK, {i3 g2 tE=1s tA=1s {M042770 SZ=4096 L=3 1001:1002 B8*4096 i0:1151492 A 1572815646.215696 M 1572815646.135696 C 1572815646.135696}}
2019/11/03 22:24:12 rx 4: LOOKUP i3 ["fichier_louis_1.7-2.txt"] 24b
FS.GetAttr("ag/fichier_louis_1.7-2.txt")
2019/11/03 22:24:12 tx 4:     OK, {i0 g0 tE=1s tA=0s {M00 SZ=0 L=0 0:0 B0*0 i0:0 A 0.000000 M 0.000000 C 0.000000}}
2019/11/03 22:24:12 rx 5: CREATE i3 {0100660 [WRONLY,CREAT,LARGEFILE,NONBLOCK] (07)} ["fichier_louis_1.7-2.txt"] 24b
2019/11/03 22:24:12 tx 5:     13=permission denied, {i0 g0 {M00 SZ=0 L=0 0:0 B0*0 i0:0 A 0.000000 M 0.000000 C 0.000000} &{0 0 0}}


$ touch /NAS/VISIBLE/donnes_gocryptfs/louis/fichier_louis_1.7-2.txt

2019/11/03 22:25:29 rx 11: GETATTR i1 {Fh 0}
FS.GetAttr("")
2019/11/03 22:25:29 tx 11:     OK, {tA=1s {M042770 SZ=4096 L=7 1000:1001 B8*4096 i0:1151489 A 1572816037.663566 M 1572816036.633572 C 1572816036.633572}}
2019/11/03 22:25:29 rx 12: LOOKUP i1 ["louis"] 6b
FS.GetAttr("louis")
2019/11/03 22:25:29 tx 12:     OK, {i4 g3 tE=1s tA=1s {M042770 SZ=4096 L=2 1002:1002 B8*4096 i0:1151502 A 1572816112.113162 M 1572816321.222024 C 1572816321.222024}}
2019/11/03 22:25:29 rx 13: LOOKUP i4 ["fichier_louis_1.7-2.txt"] 24b
FS.GetAttr("louis/fichier_louis_1.7-2.txt")
2019/11/03 22:25:29 tx 13:     OK, {i0 g0 tE=1s tA=0s {M00 SZ=0 L=0 0:0 B0*0 i0:0 A 0.000000 M 0.000000 C 0.000000}}
2019/11/03 22:25:29 rx 14: CREATE i4 {0100660 [CREAT,LARGEFILE,WRONLY,NONBLOCK] (07)} ["fichier_louis_1.7-2.txt"] 24b
FS.GetAttr("louis/fichier_louis_1.7-2.txt")
2019/11/03 22:25:29 tx 14:     OK, {i5 g5 {M0100660 SZ=0 L=1 1002:1002 B0*4096 i0:1151521 A 1572816329.441979 M 1572816329.441979 C 1572816329.441979} &{2 0 0}}
2019/11/03 22:25:29 rx 15: FLUSH i5 {Fh 2}
2019/11/03 22:25:29 tx 15:     OK
2019/11/03 22:25:29 rx 16: SETATTR i5 {atime 1572816329.441979303, mtime 1572816329.441979303}
file.GetAttr()
2019/11/03 22:25:29 tx 16:     OK, {tA=1s {M0100660 SZ=0 L=1 1002:1002 B0*4096 i0:1151521 A 1572816329.455839 M 1572816329.455842 C 1572816329.441979}}
2019/11/03 22:25:29 rx 17: FLUSH i5 {Fh 2}
2019/11/03 22:25:29 tx 17:     OK
2019/11/03 22:25:29 rx 18: RELEASE i5 {Fh 2 LARGEFILE,WRONLY,NONBLOCK  L0}
2019/11/03 22:25:29 tx 18:     OK
```

**Comment by slackner** [comment]
Please try upgrading to version 1.7.1. This sounds very much like bug https://github.com/rfjakob/gocryptfs/issues/394, that was fixed with commit https://github.com/rfjakob/gocryptfs/commit/a97d14c42d902e53efdb42e9b32c4a39559fc262.

Some more background: Before version 1.7, the files and directories were always created as root, and then assigned to the real owner. This had several disadvantages, including race conditions and the problem with sticky bit inheritance. With version 1.7 gocryptfs switched to a more robust approach, and properly impersonates the user that is trying to do a file operation. Unfortunately, this lead exactly to the regression you encountered. Only with >= 1.7.1 (or manually backporting the commit above) the secondary groups are taken into account.

**Comment by Mamak2000** [comment]
Hello,

Thank you for pointing that .deb package available was not the latest one. I thought 1.7-2 was a fancy name and that it included the changes made in 1.7.1...

So since I did not find any pre-compiled gocryptfs-1.7.1 .deb for armhf (neither for any x86) in debian packages, I eventually managed to compile it manually from the source package file gocryptfs_v1.7.1_src-deps.tar.gz (the process given on page https://nuetzlich.net/gocryptfs/compile/ did not work at first because it was looking for hanwen/go-fuse/v2/fuse and I figured out later only how to handle it :p)

Anyway, things work pretty fine now. I have gocryptfs and gocryptfs-xray binaries :) And secondary groups and sticky bits are enforced 👍 

```
pi@raspnas:~ $ sudo ls -al /NAS/VISIBLE/donnes_gocryptfs/ag
total 20
drwxrws--- 3 ag    parents 4096 nov.   4 21:57 .
drwxrws--- 7 pi    famille 4096 nov.   3 22:20 ..
-rw-rw-r-- 1 ag    parents    0 oct.  30 18:35 fichier_ag_20191030.txt
-rw-rw---- 1 ag    parents   50 nov.   3 21:01 fichier_ag.txt
-rw-rw---- 1 louis famille    0 nov.   3 22:29 fichier_louis_1.6.1.txt
-rw-rw-r-- 1 louis famille    0 nov.   3 22:13 fichier_louis_go_crypt_1.6.1.txt
-rw-rw---- 1 louis parents    0 nov.   4 21:57 fichier_louis.txt_gocrypt1.7.1 <== YEAH
```

**Comment by slackner** [comment]
Glad that it works now :)

What was the issue you encountered with the instructions provided on the gocryptfs website? If some information is outdated there (or a certain workaround is required depending on the system or Go version), feel free to open another issue in the [gocryptfs-website repository](https://github.com/rfjakob/gocryptfs-website) so that we can update/extend the instructions.

@lechner: Is there already a schedule when 1.7.1 will be available in the Debian repos?
